lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -191,8 +191,9 @@ tokenize([$# | String], Line, Column, Scope, Tokens) ->
 tokenize([$~, S, H, H, H | T] = Original, Line, Column, Scope, Tokens) when ?is_quote(H), ?is_upcase(S) orelse ?is_downcase(S) ->
   case extract_heredoc_with_interpolation(Line, Column, Scope, ?is_downcase(S), T, H) of
     {ok, NewLine, NewColumn, Parts, Rest} ->
+      Indent = NewColumn - 4,
       {Final, Modifiers} = collect_modifiers(Rest, []),
-      Token = {sigil, {Line, Column, nil}, S, Parts, Modifiers, <<H, H, H>>},
+      Token = {sigil, {Line, Column, Indent, nil}, S, Parts, Modifiers, <<H, H, H>>},
```

Comment:
I would always include the indentation. The reason is that we want the indentation even if columns is disabled otherwise EEx.SyntaxError would be incorrect. So I would remove it from the location tuple and move it to the `sigil` tuple, right after `Modifiers` (I think the last element has to be a binary) .